cvmfs/download.h

```diff
@@ -165,7 +167,11 @@ struct JobInfo {
   {
     Init();
     url = u;
-    compressed = c;
+    //compressed = c;
+    if (c)
```

Comment:
ternary operator? 
```C++
compression_alg = c ? zlib::kZlibDefault : zlib::kNoCompression
```
Or even better, make it part of the `Init()` function to avoid code duplication.